skyportal/models.py

```diff
@@ -483,7 +484,33 @@ def get_obj_if_owned_by(obj_id, user_or_token, options=[]):
 
 
 def get_obj_comments_owned_by(self, user_or_token):
-    return [comment for comment in self.comments if comment.is_owned_by(user_or_token)]
+    owned_comments = [
+        comment for comment in self.comments if comment.is_owned_by(user_or_token)
+    ]
+
+    # Grab basic author info for the comments
+    for comment in owned_comments:
+        author = User.query.filter(User.username == comment.author).first()
+        if author:
+            comment.author_info = {
+                "username": author.username,
+                "first_name": author.first_name,
+                "last_name": author.last_name,
+                "gravatar_url": author.gravatar_url,
+            }
+        else:
+            username = "unknown user"
+            digest = md5(username.lower().encode('utf-8')).hexdigest()
+            # return a 404 status code if not found on gravatar
+            gravatar_url = f'https://www.gravatar.com/avatar/{digest}?d=404'
```

Comment:
Is there a way to leverage the baselayer method in the `User` model that does this? I worry about repeat functionality here.

Reply:
Good point. I managed to do it with this [commit](https://github.com/skyportal/skyportal/pull/715/commits/6d1265b375fdcfb4ff17f32861ce1c086b9a29c2).